### ISSUE TITLE
sia waste cpu/mem by erusure code same files over and over failing to upload

### ISSUE THREAD

**Issue by vevarm** [NebulousLabs/Sia#1607]
It seems sia get stuck in some loop erasurecoding same files over and over. Wasting cpu and memory, 

We have confirmed cpu is used for erasure coding by profiling and we can see sia is reading a few source files over and over. 

This was discussed in slack contributror channel.

**Comment by DavidVorick** [comment]
I have two thoughts for this. The first is that Sia pretty much abandons efficiency on the final 0.5x redundancy, and this is probably half of it.

The second is that you are repeatedly uploading to bad hosts, and that's just an issue with the host selection algorithm.

Feel free to investigate more and submit pull requests. My priority here I think is to upgrade the contractor to swap out bad hosts, that will happen after I finish some modifications I'm making to the host

**Comment by vevarm** [comment]
Propose to close this one, this wasteful behaviour seems gone in 1.2.0.

**Comment by DavidVorick** [comment]
That's really interesting. We didn't fix this behavior in v1.2.0.

It seems that the wastefulness was caused by something else that we fixed. We did fix a handful of problematic networking code, there's a chance that crappy networking was causing problems.

Glad it's fixed, but going to leave this issue open as we still definitely do a wasteful amount of erasure coding and encryption, and it won't be that hard to fix that up and make it better.
